### ISSUE TITLE
[rllib] What happens to the replay buffer upon checkpoint restore? 

### ISSUE THREAD

**Issue by regproj** [ray-project/ray#8151]
### What is your question?
I've been having to restart training sessions (using td3, sac, whatever) and was wondering what happens to the replay buffer when restoring from a checkpoint. Is the replay buffer now empty?
*Ray version and other system information (Python version, TensorFlow version, OS):*
Ray 0.8.4
Python 3.6
Tf 1.12.0
Ubuntu 18.04

**Comment by anguyenbus** [comment]
Second to @regproj

**Comment by cedros23** [comment]
Hello, I am curious about the answer too. 
As I now try to restore Apex-DQN agent, it appears that agent was initially restored successfully with the correct number of iteration and on par performance but immediately got worse after few iterations. Almost forgot all the skills it learnt.

**Comment by MischaPanch** [comment]
A comment just to revive this issue. The current status is that after restoring an agent the buffer is gone. It would be really helpful for me to be able to restore the buffer since I want to pretrain an SAC agent on a modified reward. Without reloading the replay buffer the training begins by forgetting what was learned (although in my case the agent recovers pretty fast), see image below. There was a similar issue [here](https://github.com/ray-project/ray/issues/2330) that was closed saying that reloading buffers would be implemented eventually. I cannot find documentation on it, could you maybe point me to it? If this was not implemented, I would be happy to try and make a PR solving this issue.

Here is how the mean reward typically looks if you resume the training with an empty buffer

![image](https://user-images.githubusercontent.com/35432522/109501347-c2b19380-7a97-11eb-9eb7-99cebbeb45e4.png)

**Comment by mvindiola1** [comment]
As far as I know this is still an issue

**Comment by sungho-joo** [comment]
This is also an important issue for me as well.
Is there any update for this issue?
Seems like storing replay buffer in checkpoint is in progress

**Comment by sven1977** [comment]
Hey @sungho-joo , actually, here is a PR that's currently in review, which will store the contents of the replay buffer to a checkpoint, if some config flag is set to True. The data is then also reloaded properly. Should get merged soon:

https://github.com/ray-project/ray/pull/17999

**Comment by sungho-joo** [comment]
Thanks for replying @sven1977. Actually I already looked over the PR and was waiting for the merge. Cheers!
